### DISCUSSION TITLE
Outputting Kernels

### DISCUSSION THREAD

**Discussion by bbattas** [idaholab/moose#28058] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I had thought there was a way to basically print out all the kernels used in MOOSE instead of solving, but I can't find it in the documentation anywhere.  I want to see all the individual kernels created by an action, and ideally also their various components like would be in an input file if I had manually defined the kernel.  Is there a way to do that?

**Comment by GiudGiud** [comment, marked as answer]
Hello

There are debugging options listed there.
Debug/show_actions is probably one of the easiest ways to do this
https://mooseframework.inl.gov/source/actions/SetupDebugAction.html

otherwise:
```
[Problem]
  verbose_setup = true
```
is helpful and so is switching to this problem:
https://mooseframework.inl.gov/source/problems/DumpObjectsProblem.html
